--- gnutls28.orig/lib/nettle/rnd-linux.c
+++ gnutls28/lib/nettle/rnd-linux.c
@@ -51,13 +51,15 @@ static ino_t _gnutls_urandom_fd_ino = 0;
 static dev_t _gnutls_urandom_fd_rdev = 0;
 
 #if defined(__linux__)
-# ifdef HAVE_LINUX_GETRANDOM
-#  include <linux/random.h>
-# else
-#  include <sys/syscall.h>
-#  undef getrandom
-#  define getrandom(dst,s,flags) syscall(SYS_getrandom, (void*)dst, (size_t)s, (unsigned int)flags)
-# endif
+# include <linux/version.h>
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(3,17,0)
+#  ifdef HAVE_LINUX_GETRANDOM
+#   include <linux/random.h>
+#  else
+#   include <sys/syscall.h>
+#   undef getrandom
+#   define getrandom(dst,s,flags) syscall(SYS_getrandom, (void*)dst, (size_t)s, (unsigned int)flags)
+#  endif
 
 static unsigned have_getrandom(void)
 {
@@ -107,6 +109,9 @@ static int _rnd_get_system_entropy_getra
 
 	return 0;
 }
+# else /* not linux version >= 3.4.17 */
+#  define have_getrandom() 0
+# endif
 #else /* not linux */
 # define have_getrandom() 0
 #endif
@@ -162,7 +167,7 @@ int _rnd_system_entropy_init(void)
 	int old;
 	struct stat st;
 
-#if defined(__linux__)
+#if defined(__linux__) && LINUX_VERSION_CODE >= KERNEL_VERSION(3,17,0)
 	/* Enable getrandom() usage if available */
 	if (have_getrandom()) {
 		_rnd_get_system_entropy = _rnd_get_system_entropy_getrandom;
